libcudacxx/include/cuda/__iterator/transform_input_output_iterator.h

```diff
@@ -0,0 +1,492 @@
+  using iterator_concept = iterator_category;
+  using difference_type  = _CUDA_VSTD::iter_difference_t<_Iter>;
+  using value_type       = _CUDA_VSTD::invoke_result_t<_InputFn, _CUDA_VSTD::iter_value_t<_Iter>>;
+  using pointer          = void;
+  using reference        = __transform_input_output_proxy<_Iter, _InputFn, _OutputFn>;
+
+  //! @brief Default constructs a \p transform_input_output_iterator with a value initialized iterator and functors
+#if _CCCL_HAS_CONCEPTS()
+  _CCCL_EXEC_CHECK_DISABLE
+  _CCCL_HIDE_FROM_ABI transform_input_output_iterator()
+    requires _CUDA_VSTD::default_initializable<_Iter> && _CUDA_VSTD::default_initializable<_InputFn>
+            && _CUDA_VSTD::default_initializable<_OutputFn>
+  = default;
+#else // ^^^ _CCCL_HAS_CONCEPTS() ^^^ / vvv !_CCCL_HAS_CONCEPTS() vvv
+  _CCCL_EXEC_CHECK_DISABLE
+  _CCCL_TEMPLATE(class _Iter2 = _Iter, class _InputFn2 = _InputFn, class _OutputFn2 = _OutputFn)
+  _CCCL_REQUIRES(_CUDA_VSTD::default_initializable<_Iter2> _CCCL_AND
+                   _CUDA_VSTD::default_initializable<_InputFn2> _CCCL_AND _CUDA_VSTD::default_initializable<_OutputFn2>)
+  _CCCL_API constexpr transform_input_output_iterator() noexcept(
+    _CUDA_VSTD::is_nothrow_default_constructible_v<_Iter2> && _CUDA_VSTD::is_nothrow_default_constructible_v<_InputFn2>
+    && _CUDA_VSTD::is_nothrow_default_constructible_v<_OutputFn2>)
+  {}
+#endif // ^^^ !_CCCL_HAS_CONCEPTS() ^^^
+
+  //! @brief Constructs a \p transform_input_output_iterator with a given \p __iter iterator, \p __input_func and
+  //! \p __output_func functors
+  //! @param __iter The iterator to transform
+  //! @param __input_func The functor to apply to the iterator when reading
+  //! @param __output_func The functor to apply to the iterator when writing
+  _CCCL_EXEC_CHECK_DISABLE
+  _CCCL_API constexpr transform_input_output_iterator(
+    _Iter __current,
+    _InputFn __input_func,
+    _OutputFn __output_func) noexcept(_CUDA_VSTD::is_nothrow_move_constructible_v<_Iter>
+                                      && _CUDA_VSTD::is_nothrow_move_constructible_v<_InputFn>
+                                      && _CUDA_VSTD::is_nothrow_move_constructible_v<_OutputFn>)
+      : __current_(_CUDA_VSTD::move(__current))
+      , __input_func_(_CUDA_VSTD::in_place, _CUDA_VSTD::move(__input_func))
+      , __output_func_(_CUDA_VSTD::in_place, _CUDA_VSTD::move(__output_func))
+  {}
+
+  //! @brief Returns a const reference to the iterator stored in this \p transform_input_output_iterator
+  [[nodiscard]] _CCCL_API constexpr const _Iter& base() const& noexcept
+  {
+    return __current_;
+  }
+
+  //! @brief Extracts the iterator stored in this \p transform_input_output_iterator
+  _CCCL_EXEC_CHECK_DISABLE
+  [[nodiscard]] _CCCL_API constexpr _Iter base() && noexcept(_CUDA_VSTD::is_nothrow_move_constructible_v<_Iter>)
+  {
+    return _CUDA_VSTD::move(__current_);
+  }
+
+  //! @brief Returns a proxy that transforms the input upon assignment
```

Comment:
These descriptions were not yet updated (coming from `transform_output_iterator`). Possible formulation:
```suggestion
  //! @brief Returns a proxy that transforms read values via `_InputFn` upon converting to `value_type` and transforms assigned values via `_OutputFn` before writing
```